internal/jimmhttp/httpproxy_handler.go

```diff
@@ -55,23 +54,24 @@ func (hph *HTTPProxyHandler) ProxyHTTP(w http.ResponseWriter, req *http.Request)
 
 	modelUUID := chi.URLParam(req, "uuid")
 	if modelUUID == "" {
-		writeError(ctx, w, http.StatusUnprocessableEntity, errgo.New("cannot parse path"), "cannot parse path")
+		writeError(ctx, w, http.StatusUnprocessableEntity, errors.E("cannot parse path"), "cannot parse path")
 		return
 	}
-	model, err := hph.jimm.JujuManager().GetModel(ctx, modelUUID)
-	if err != nil {
-		writeError(ctx, w, http.StatusNotFound, err, "cannot get model")
-		return
-	}
-	u, p, err := hph.jimm.CredentialStore.GetControllerCredentials(ctx, model.Controller.Name)
+
```

Comment:
Should validate model uuid here, no reason to continue and call controller details if the uuid is invalid.